api/institutions/authentication.py

```diff
@@ -102,6 +104,12 @@ def authenticate(self, request):
         # unconfirmed, etc.).
         user, created = get_or_create_user(fullname, username, reset_password=False)
 
+        # The `department` field is updated each login, with multiple departments first in wins.
+        department = department or departments[0]
```

Comment:
If we are handling this rather than CAS, then we have to sort the departments, or it could potentially change every login.

Reply:
As mentioned, let CAS handle the sort and selection. The raw value is something like

`ou=Music Department, o=Notre Dame, dc=nd, dc=edu`

and CAS needs to parse it out anyway.